### PR TITLE
Added support for winds.el

### PR DESCRIPTION
**Pull request by CsBigDataHub**
Added support for [winds.el ](https://github.com/Javyre/winds.el)

### CODE REVIEW

File: doom-modeline-segments.el [hunk 1]
```diff
@@ -1353,30 +1353,34 @@ one. The ignored buffers are excluded unless `aw-ignore-on' is nil."
...
-                    'face (if (doom-modeline--active)
-                              'doom-modeline-buffer-major-mode
-                            'mode-line-inactive)))))
+(doom-modeline-def-segment workspace-name
+  "The current workspace name or number.
+Requires `winds.el' to be loaded, or `eyebrowse-mode' or
+`Requires tab-bar-mode' to be enabled."
+  (when-let
+      ((name (cond
+              ((featurep 'winds)
```
Comment (seagle0128) [inline doom-modeline-segments.el:1362]: This doesn't mean winds is enabled.

**Comment by seagle0128** [comment]
Please provide screenshot, and clear all compiler warnings.

File: doom-modeline-segments.el (hunk 1 above)
Comment (CsBigDataHub) [reply doom-modeline-segments.el:1362]: Sorry , Do I need to use `bound-and-true-p`. I am not sure.

**Comment by CsBigDataHub** [comment]
screenhots - 

image

image

With flycheck this is all I am getting,
` doom-m…  2650   1 warning         the following functions are not known to be defined: winds-get-cur-ws, winds-get-cur-cfg, seq-contains-p (emacs-lisp)`

**Comment by seagle0128** [comment]
> With flycheck this is all I am getting,
doom-m… 2650 1 warning the following functions are not known to be defined: winds-get-cur-ws, winds-get-cur-cfg, seq-contains-p (emacs-lisp)

You need to declare the external functions to clean the warnings., as

https://github.com/seagle0128/doom-modeline/blob/master/doom-modeline-segments.el#L160

File: doom-modeline-segments.el (hunk 1 above)
Comment (seagle0128) [reply doom-modeline-segments.el]: I don't know how to check if it's enabled. What I concern is this case: if installed both `winds` and `eyebrowse`, but only enabled `eyebrowse`. These codes are incorrect to check. 

BTW, it's not released on melpa, right? What's the advantage of `winds` comparing to `eyebrowse`?

**Comment by CsBigDataHub** [comment]
Wins is released in melpa - https://melpa.org/#/winds. 

This is a post from winds author . He goes through the difference between winds and eyebrowse - [here](https://www.reddit.com/r/emacs/comments/g3s0ts/windsel_2dimensional_workspace_switcher/?utm_medium=android_app&utm_source=share)

**Comment by seagle0128** [comment]
Any update for this PR? @CsBigDataHub 
Will you address the issues above, or close this?

**Comment by CsBigDataHub** [comment]
> Any update for this PR? @CsBigDataHub
> Will you address the issues above, or close this?

@seagle0128 
Sorry could not get around to it. I got a note from the author that he is in the process of enhancing winds.el . https://github.com/Javyre/winds.el/issues/1#issuecomment-623494245

Could you please leave it open for a little while. I will get around to it.

Thanks

**Comment by seagle0128** [comment]
Okay. Then let's leave it open.

**Comment by Javyre** [comment]
`winds` now has a `winds-mode` you have to activate so the `winds-mode` boolean now exists for properly checking if winds is enabled. `bound-and-true-p` should work fine.

File: doom-modeline-segments.el [hunk 2]
```diff
@@ -1353,30 +1353,34 @@ one. The ignored buffers are excluded unless `aw-ignore-on' is nil."
 (advice-add #'winum--install-mode-line :override #'ignore)
 (advice-add #'winum--clear-mode-line :override #'ignore)
 
-(doom-modeline-def-segment window-number
```
Comment (seagle0128) [inline doom-modeline-segments.el:1356]: Why delete window-number segment?

File: doom-modeline-segments.el [hunk 3]
```diff
@@ -1353,30 +1353,34 @@ one. The ignored buffers are excluded unless `aw-ignore-on' is nil."
...
-                    'face (if (doom-modeline--active)
-                              'doom-modeline-buffer-major-mode
-                            'mode-line-inactive)))))
+(doom-modeline-def-segment workspace-name
+  "The current workspace name or number.
+Requires `winds.el' to be loaded, or `eyebrowse-mode' or
+`Requires tab-bar-mode' to be enabled."
+  (when-let
+      ((name (cond
+              ((bound-and-true-p winds)
```
Comment (seagle0128) [inline doom-modeline-segments.el:1362]: do you mean `winds-mode`?

**Review by seagle0128 — CHANGES_REQUESTED** [review]

**Comment by CsBigDataHub** [comment]
@seagle0128 Let me look in to this during thanksgiving weekend. 

Thanks @Javyre

**Comment by CsBigDataHub** [comment]
I am closing this PR, I have been using eyebrowse  with persp-mode and it gives me nearly the same functionality of winds.el

**Comment by Javyre** [comment]
@CsBigDataHub may I ask what it is that drew you away from using `winds.el`?

I think this PR still has value in that it would help others smoothly transition to `winds.el` with less hassle on doom.

**Comment by CsBigDataHub** [comment]
> @CsBigDataHub may I ask what it is that drew you away from using `winds.el`?
> 
> I think this PR still has value in that it would help others smoothly transition to `winds.el` with less hassle on doom. 

@Javyre I have just tried `winds.el` for few days, but I have soon noticed that using `prep-mode` gives me the same functionality of `winds.el` when used together with `eyebrowse`. More over `doom-mode-line` gives me indication of my perspective name and the eyebrowse window number. 

So I have a perspective for each project which has multiple window configurations.

Yes I agree that this PR has value but I cannot work on it. I am pretty much tied up with my personal and professional commitments. I can reopen it if you are willing to contribute.

Thanks.

**Pull request closed without merging**